Resources/Prototypes/_NF/Catalog/Fills/Crates/salvage.yml

```diff
@@ -4,308 +4,60 @@
   categories: [ HideSpawnMenu ]
   parent: CrateGenericSteel
   components:
-  - type: StorageFill
-    contents:
-    - id: OxygenTankFilled
-      prob: 0.1
-      orGroup: Gas
-    - id: NitrogenTankFilled
-      prob: 0.1
-      orGroup: Gas
-    - prob: 2.0 # Empty
-      orGroup: Gas
-   # - id: ResearchDisk
-    #  prob: 0.1
-    #  orGroup: RD
-    - id: ResearchDisk5000
-      prob: 0.01
-      orGroup: RD
-    - id: ResearchDisk10000
-      prob: 0.005
-      orGroup: RD
-    - id: ResearchDisk10000
-      prob: 0.0025
-      orGroup: RD
-    - prob: 2 # Empty
-      orGroup: RD
-    # Normal (10%)
+  - type: EntityTableContainerFill
+    containers:
+      entity_storage: !type:NestedSelector
+        tableId: NFSalvageAssortedGoodies
+        rolls: !type:RangeNumberSelector
+          range: 0, 3
+
+- type: entityTable
+  id: NFSalvageAssortedGoodies
+  table: !type:GroupSelector
+    children:
+    # Mats
     - id: SheetPlasma
-      prob: 0.1
+      weight: 0.1
```

Comment:
Might recommend adding steel and glass if only to dilute this table somewhat, - it's _very valuable_ as-is, very compressable, and wrecks are _littered_ with these.

Also, the range selector is giving us 4 items - there's a 1 added to a random.NextFloat call, so you get a number between [0,4) then iterate from 0 _to_ that number - pick 3.91, iterate 0, 1, 2, 3 - four items.